runtime/src/main/kotlin/kotlin/reflect/AssociatedObjects.kt

```diff
@@ -17,9 +17,5 @@ annotation class ExperimentalAssociatedObjects
 annotation class AssociatedObjectKey
 
 @ExperimentalAssociatedObjects
-fun KClass<*>.getAssociatedObject(key: KClass<*>): Any? =
-        if (this is KClassImpl<*> && key is KClassImpl<*>) {
-            this.getAssociatedObjectImpl(key)
-        } else {
-            null
-        }
+inline fun <reified T> KClass<*>.findAssociatedObject(): Any? =
```

Comment:
Needs docs! (here and for other public declarations)

Reply:
And also explicit `public` visibility for declarations.